### PR TITLE
Update SC2 for 0.4.3

### PR DESCRIPTION
**Pull request by Ziktofel**
Updated to 0.4.3 YAML and set things for long async games accessible to new players

### CODE REVIEW

**Comment by ScootyPuffJr1** [comment]
I've verified this runs just fine, but I know nothing about the game to comment on option weights.

**Comment by Ziktofel** [comment]
I'm the game's current maintainer so I know about these new stuff

The idea is to for full game enable all items and do reduction by sets only if a shorter campaign is in
For other settings I've picked the most popular options while still being friendly to new players (therefore no default Brutals)

File: games/Starcraft 2 Wings of Liberty.yaml [hunk 1]
```diff
@@ -1,19 +1,75 @@
...
+  final_map:
+    all_in: 25
+    random_hard: 75
   shuffle_protoss: random
   shuffle_no_build: random
   early_unit: true
   required_tactics: standard
-  units_always_have_upgrades: true
\ No newline at end of file
+  units_always_have_upgrades: true
```
Comment (t3hf1gm3nt) [inline games/Starcraft 2 Wings of Liberty.yaml:20]: just wondering if there should be a chance for this to be false? not saying it has to but would like some discussion about it

File: games/Starcraft 2 Wings of Liberty.yaml [hunk 2]
```diff
@@ -1,19 +1,75 @@
...
+  generic_upgrade_research:
+    auto_in_no_build: 75
+    always_auto: 25
+  nco_items: true
+  bw_items: true
+  ext_items: true
```
Comment (t3hf1gm3nt) [inline games/Starcraft 2 Wings of Liberty.yaml:26]: i did notice the triggers at the bottom that makes these have a chance to be false on all the mission orders besides pure vanilla. wouldn't it be smarter to put the chances here and have one trigger for vanilla to turn them all on? on that note, why the special exception for pure vanilla?

File: games/Starcraft 2 Wings of Liberty.yaml [hunk 3]
```diff
@@ -1,19 +1,75 @@
...
+  mission_progress_locations: enabled
+  bonus_locations: enabled
+  challenge_locations:
+    enabled: 75
+    trash: 25
```
Comment (t3hf1gm3nt) [inline games/Starcraft 2 Wings of Liberty.yaml:31]: i'd 50/50 this personally. interested in your reasoning behind the 3/4 split

File: games/Starcraft 2 Wings of Liberty.yaml (hunk 2 above)
Comment (Ziktofel) [reply games/Starcraft 2 Wings of Liberty.yaml:26]: Vanilla shuffled uses always on too, it's to not generate too many resources to the player

File: games/Starcraft 2 Wings of Liberty.yaml (hunk 3 above)
Comment (Ziktofel) [reply games/Starcraft 2 Wings of Liberty.yaml:31]: I think that most players want to have them enabled (the most usual setting) therefore this split

File: games/Starcraft 2 Wings of Liberty.yaml (hunk 1 above)
Comment (Ziktofel) [reply games/Starcraft 2 Wings of Liberty.yaml:20]: This setting to false is usually used on shorter games, therefore I kept that on

**Pull request merged**